src/main/java/com/google/cloud/teleport/splunk/SplunkEventWriter.java

```diff
@@ -264,16 +266,35 @@ private void logWriteFailures(@StateId(COUNT_STATE_NAME) ValueState<Long> countS
     LOG.error("Failed to write {} events", countState.read());
     FAILED_WRITES.inc(countState.read());
   }
+  
+  /**
+   * Utility method to flush failed write events.
+   *
+   * @param builder {@link SplunkWriteError.Builder} object
+   * @param events List of {@link SplunkEvent}s to un-batch
+   * @param receiver Receiver to write {@link SplunkWriteError}s to
+   */
+  private void flushWriteFailures(
+      SplunkWriteError.Builder builder,
+      List<SplunkEvent> events,
+      OutputReceiver<SplunkWriteError> receiver) {
+    
+    for (SplunkEvent event : events) {
+      String payload = GSON.toJson(event);
+      SplunkWriteError error = builder.withPayload(payload).build();
+      
+      receiver.output(error);
+    }
+  }
 
   /**
    * A helper method to check the {@link HttpResponse} object and wrap the response into an {@link
    * Optional} {@link SplunkWriteError} in case of an unsuccessful response.
    *
    * @param response {@link HttpResponse} object
-   * @return {@link Optional} {@link SplunkWriteError}
+   * @return {@link Optional} {@link SplunkWriteError.Builder}
    */
-  private Optional<SplunkWriteError> checkResponse(HttpResponse response)
-      throws IOException {
+  private Optional<SplunkWriteError.Builder> checkResponse(HttpResponse response) {
```

Comment:
I prefer to not return a builder here. What else can we return to make this cleaner? We basically need the statusCode and statusMessage from the response if it is unsuccessful.

Reply:
We can check the HttpResponse success status code inside the existing `flush` method itself and call flushWriteFailures to build and output failed events based on the success status code. We won't need the `checkResponse` method then.